juju/juju
# Remove the fan info field from subnet structs

As part of the drop of fan networking, this patch removes the `FanInfo` field (and associated structs) from the `SubnetInfo` struct.

Tests and facades that referenced it were fixed or removed depending the case.

## Checklist


## QA steps

This is mainly removal of unused code at this point, but as QA steps, (just like the previous PR https://github.com/juju/juju/pull/17429) bootstrapping on AWS and not listing fan subnets should still work:
```
juju bootstrap aws c
juju add-model m
juju subnets
subnets:
  172.31.0.0/20:
    type: ipv4
    provider-id: subnet-7d4c9614
    provider-network-id: vpc-8abf64e3
    space: alpha
    zones:
    - eu-west-3a
  172.31.16.0/20:
    type: ipv4
    provider-id: subnet-8641bdfd
    provider-network-id: vpc-8abf64e3
    space: alpha
    zones:
    - eu-west-3b
  172.31.32.0/20:
    type: ipv4
    provider-id: subnet-748c9d3e
    provider-network-id: vpc-8abf64e3
    space: alpha
    zones:
    - eu-west-3c
  172.31.254.0/24:
    type: ipv4
    provider-id: subnet-034e151128f81628c
    provider-network-id: vpc-8abf64e3
    space: alpha
    zones:
    - eu-west-3c
```
Also, to make sure the network config is correctly set by the container init in the provisioner worker, deploying to a container should continue working:
```
juju deploy ubuntu --to lxd
```

Make sure 3.6->4.0 migration isn't broken:
```
juju_36 bootstrap aws 36
juju_36 add-model m
juju_36 deploy ubuntu --to lxd
# install this patch at this point
juju bootstrap aws c40
juju switch c36:admin/m
juju migrate c40
```
## Links

**Jira card:** JUJU-6050
